pkgs/build-support/build-typst-package.nix

```
@@ -0,0 +1,31 @@
+{ lib, stdenvNoCC }:
+{
+  nativeBuildInputs ? [ ],
+  buildInputs ? [ ],
+  meta ? { },
+  ...
+}@attrs:
+let
+  cleanAttrs = lib.flip lib.removeAttrs [ ];
+  typstPkgPrefix = "typst-package";
+in
+stdenvNoCC.mkDerivation (
+  (cleanAttrs attrs)
+  // (with attrs; {
+    name = lib.concatStringsSep "-" [
+      typstPkgPrefix
+      pname
+      version
+    ];
+    installPhase =
+      let
+        outDir = "$out/${pname}/${version}";
+      in
+      ''
+        ${attrs.installPhase}
```

```suggestion
        runHook preInstall
```
Usually, setting `installPhase` in packages overrides the entirety of the builder's default `installPhase`. This should be replaced by a `preInstall` hook.

---

This could even be its own hook that runs after `postInstall`, but before the `fixupPhase`.